components/net/hsts.rs

```diff
@@ -139,15 +142,160 @@ impl HstsList {
     }
 
     /// Step 10 of https://fetch.spec.whatwg.org/#concept-main-fetch.
-    pub fn switch_known_hsts_host_domain_url_to_https(&self, url: &mut ServoUrl) {
-        if url.scheme() != "http" {
+    pub fn apply_hsts_rules(&self, url: &mut ServoUrl) {
+        if url.scheme() != "http" && url.scheme() != "ws" {
```

Comment:
Is there a spec link for rewriting ws as wss?

Reply:
https://fetch.spec.whatwg.org/#websocket-opening-handshake
> 1. Let requestURL be a copy of url, with its scheme set to "http", if url’s scheme is "ws", and to "https" otherwise.
> 
> This change of scheme is essential to integrate well with fetching. E.g., HSTS would not work without it. There is no real reason for WebSocket to have distinct schemes, it’s a legacy artefact. [HSTS]

The spec recommends to work with a http(s) url, but ws-rs expects to get a ws(s) url. I did not change this aspect.
https://github.com/servo/servo/pull/25404/commits/6dc1f49c6eee059e6bfb8bcbbc53239a7d6f18c0#diff-68b5c651915d8b239e5e3b6c84f02981L194-R252
net_url contains the `ws(s)://` url as it is needed for ws-rs. Instead of converting the upgraded url back to wss in websocket_loader.rs, I kept it clean and upgraded ws to wss.
#14897 seems to be about changing the websocket code to be tighter along the spec.

Reply:
Can you add that link (and the discussion about ws urls) to your PR?

Reply:
Added here: https://github.com/servo/servo/pull/25404/files#diff-68b5c651915d8b239e5e3b6c84f02981R202